source/Cosmos.HAL2/PS2Mouse.cs

```diff
@@ -131,12 +131,15 @@ public void HandleMouse(ref INTs.IRQContext context)
                 {
                     mouseCycle++;
                 }
+            }else if (mouseCycle == 3) {
+                mouseByte[3] = IOPort.Read8(Core.IOGroup.PS2Controller.Data);
+                mouseCycle++;
             }
 
             // TODO: move conditions to the if statement when stack corruption detection
             //       works better for complex conditions
             var xTest1 = mouseCycle == 2 && !HasScrollWheel;
-            var xTest2 = mouseCycle == 3 && HasScrollWheel;
+            var xTest2 = mouseCycle == 4 && HasScrollWheel;
 
             if (xTest1 || xTest2)
```

Comment:
We can get rid of these helper bools
```suggestion
            if ((mouseCycle == 2 && !HasScrollWheel) || (mouseCycle == 4 && HasScrollWheel))
```